cmd/gitserver/internal/server_grpc.go

```diff
@@ -1383,6 +1383,48 @@ func (gs *grpcServer) FirstEverCommit(ctx context.Context, request *proto.FirstE
 				Repo: request.GetRepoName(),
 				Spec: revisionErr.Spec,
 			})
+			if err != nil {
+				return nil, s.Err()
+			}
+
+			gs.svc.LogIfCorrupt(ctx, repoName, err)
+			return nil, err
+		}
```

Comment:
I tihnk there's a bug here:

```suggestion
			if err != nil {
				return nil, err // I think s would be potentially nil here if this op errored
			}
			
			return nil, s.Err() // return s.Err() if it's the well-known error
		}

// otherwise always also return the error, this only returns the error when s is of type revnotfounderr
		gs.svc.LogIfCorrupt(ctx, repoName, err)
		return nil, err
```